gems/aws-sdk-core/lib/aws-sdk-core/credentials_provider_chain.rb

```diff
@@ -9,16 +9,18 @@ def initialize(config)
     end
 
     def identity(_properties = {})
-      return @identity if @identity
+      # TODO: cache the provider based on config keys - so that operation
```

Comment:
I'm still on the fence on this one and not entirely convinced that we should redo default resolution based on operation config overrides.

Reply:
I think it's useful and correct to respect operation overrides for things like profile, region, credentials, etc, and we should re-resolve chain in that case.